### PR TITLE
Full main, added CrossEntropy2D, binary mask for data loader

### PR DESCRIPTION
**Pull request by raphaelattias**
1. The main is now working as intended, and we can train a model with the data loader, optimizer and loss.

2. Added Cross Entropy 2D from a random github to the loss.py file. Ultimately, our loss_function for training should be the CE2D. Right now for proof of concept we are using the wrong loss (MSE).

3. Updated the data loader, so the mask is a single channel of binary values (right now its float).

Co-Authored-By: dust629 <32882147+dust629@users.noreply.github.com>

### CODE REVIEW

File: process_data/data_loader.py [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+import torch
+
 
 class DataLoaderSegmentation(data.Dataset):
     def __init__(self, folder_path_img,folder_path_mask):
         super(DataLoaderSegmentation, self).__init__()
         self.img_files = glob.glob(os.path.join(folder_path_img,'*.png'))
         self.mask_files =glob.glob(os.path.join(folder_path_mask,'*.png'))
-        self.transforms = transforms.ToTensor()
+        self.transformsimage = transforms.Compose([transforms.ToTensor(),transforms.Resize((572,572))])
```
Comment (riccardocadei) [inline process_data/data_loader.py:15]: Why do we need to resize?

File: loss/loss.py [hunk 2]
```diff
@@ -39,4 +43,13 @@ def recall(pred,target):
     """
     TP = np.sum(np.logical_and(pred == 1, target == 1))
     FN = np.sum(np.logical_and(pred == 0, target == 1))
-    return TP/(TP + FN)
\ No newline at end of file
+    return TP/(TP + FN)
+
+
+class CrossEntropyLoss2d(nn.Module):
```
Comment (riccardocadei) [inline loss/loss.py:49]: What is the problem of this function?

File: process_data/data_loader.py (hunk 1 above)
Comment (dust629) [reply process_data/data_loader.py:15]: make it same as the size in paper

> Why do we need to resize?

File: loss/loss.py (hunk 2 above)
Comment (dust629) [reply loss/loss.py:49]: it does not work well because it cannot match different dimensions properly, I will work on it

Comment (dust629) [reply loss/loss.py:49]: > What is the problem of this function?


it does not work well because it cannot match different dimensions properly, I will work on it
